### ISSUE TITLE
Zoom Extents with specified points as bounds

### ISSUE THREAD

**Issue by saivaradula** [jumpinjackie/mapguide-react-layout#152]
Hello,

I would like to zoomExtent my map with specified points like (x1, y1) and x2, y2 as bounds from the widget ( task pane content ).
I searched in viewer.js file, but I could not find out zoomExtents with specific points as bounds. 
Viewer.js is providing full map extents. 
I am using acua in react-layout .

Any help ??

**Comment by jumpinjackie** [comment]
Currently, you can only zoom via x/y/scale and not bounding box. What I can (and should/will) do, is open up access to the [IMapViewer](https://github.com/jumpinjackie/mapguide-react-layout/blob/master/src/api/common.ts#L252) for task pane content so that your task pane content code can do this

```
var viewer = parent.getMapViewer();
var view = viewer.getViewForExtent([x1, y1, x2, y2]);
viewer.zoomToView(view.x, view.y, view.scale);
```

How does that sound?

**Comment by saivaradula** [comment]
Hello Jackie,
That might really help in zoomExtents for user defined bounds.

Thanks,
Sai

**Comment by saivaradula** [comment]
Hello Jackie:

I am getting "getViewForExtent is not a function" error when I try to access the function.
I am using below declaration in aqua of mapguide in viewer folder. 
   var viewer = new MapGuide.Application();    // of mapguide 2017

any suggestion ?

**Comment by jumpinjackie** [comment]
This function is not currently exported to any public API, so you can't access it anywhere.

**Comment by saivaradula** [comment]
Okay, when the access is open, I will try to access. 

Sai

**Comment by jumpinjackie** [comment]
The IMapViewer API interface is now accessible from task pane content via `parent.parent.GetViewerInterface()`

I've added a demo page, that you can set up an InvokeURL command to point to the URL of this page `http://servername/mapguide/your_react_viewer_virtual_dir/examples/taskpane/viewerapi.html`

And in that demo page is an example of how to zoom to bounds

https://github.com/jumpinjackie/mapguide-react-layout/blob/master/viewer/examples/taskpane/viewerapi.html#L115

This will be available in the 0.9 release, or if you want to try this right now grab the bleeding edge `viewer.zip` from the [AppVeyor build artifacts page](https://ci.appveyor.com/project/jumpinjackie/mapguide-react-layout/build/artifacts)
